### PR TITLE
feature: Create RequestContext

### LINKED ISSUE #38: RequestContext Necessity
## Necessity

- Random/Recommend Quiz APIs are depends on __Member Identifier(like AccessToken, etc)__
- The member identifier will come in a header.
- Currently, tokens are not used, so use member IDs in headers
  - `Authorization {memberId}`

### PR DESCRIPTION
**Pull request by BAEKJungHo**
## Related Issue
close #38 

## Description

- The method(randomQuizzes) on the controller will be modified while implementing Random API.

## References

- https://projectreactor.io/docs/core/release/reference/#context.propagation

### CODE REVIEW

File: core/src/main/kotlin/team/backend/context/RequestContext.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Get Value in Context
+     * @param key key of Context
+     * @return value in Context
+     */
+    suspend fun getValue(key: String): String {
+        return Mono
+            .deferContextual { ctx: ContextView ->
+                val authorization = ctx.get<String>(key)
```
Comment (mangchhe) [inline core/src/main/kotlin/team/backend/context/RequestContext.kt:18]: The function does not appear to be dependent on authorization.
The name seems to need to be changed.

File: core/src/main/kotlin/team/backend/context/RequestContext.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Get header by key
+     * @param key header key
+     * @param exchange ServerWebExchange object
+     * @return header
+     */
+    fun getHeader(key: String, exchange: ServerWebExchange): String {
+        val request = exchange.request
+        return if (key == "Authorization") {
```
Comment (mangchhe) [inline core/src/main/kotlin/team/backend/context/RequestContext.kt:31]: Why only Authorization handles exceptions?
I think other header values will need it too.

**Review by mangchhe — APPROVED** [review]
GOOD! 👍

**Comment by BAEKJungHo** [comment]
I applied feedback.

**Pull request merged**